modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -288,6 +289,24 @@ public CompletableFuture<Set<String>> dataNodes(long causalityToken, int zoneId)
         return causalityDataNodesEngine.dataNodes(causalityToken, zoneId);
     }
 
+    /**
+     * Returns the actual data nodes of the specified zone.
+     *
+     * @param zoneId Zone id.
+     * @return The future which will be completed with data nodes for the zoneId.
+     */
+    public CompletableFuture<Set<String>> currentDataNodes(int zoneId) {
+        return metaStorageManager.get(zoneDataNodesKey(zoneId)).thenApply(dataNodesEntry -> {
```

Comment:
So, you're mixing distributively current dataNodes with locally current zone cfg, right? That seems incorrect.

Reply:
I've removed this method